tests-foundry/unit/SystemRewardTest.t.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache2.0
+pragma solidity 0.8.4;
+
+import {console} from "forge-std/console.sol";
+import {BaseTest} from "../common/BaseTest.t.sol";
+import {SystemRewardMock} from "../../contracts/mock/SystemRewardMock.sol";
+
+contract SystemRewardTest is BaseTest  {
+    string private constant IS_BURN_KEY = "isBurn";
+
+    SystemRewardMock internal s_systemReward;
+
+    event rewardTo(address indexed to, uint256 amount);
+    event rewardEmpty();
+    event receiveDeposit(address indexed from, uint256 amount);
+    event paramChange(string key, bytes value);
+
+    constructor() BaseTest(REJECT_PAYMENTS) {}
```

Comment:
Meaning of the 'reject' flag: any attempt by the tested contracts to pass eth back to this test (i.e. their mesg.sender) will revert